File: RAI/Clase16/Java/Interfaces/Post.java
//package RAI.Clase16.Java.Interfaces;

public class Post implements Texto, Imagen, Video{
    // La sub clase esta obligada a implementar TODOS los metodos de TODAS las interfaces involucradas
    private int id;
    private String url;
    private String mensaje;

    public Post(int id, String url, String mensaje){
        this.id = id;
        this.url = url;
        this.mensaje = mensaje;
    }

    @Override
    public void verTexto(){
        System.out.println("> Texto: " + this.mensaje);
    }

    @Override
    public void verImagen(){
        System.out.println("> Imagen: " + this.url);
    }
    
    @Override
    public void verVideo(){
        System.out.println("> Video: " + this.url);
    }
}

File: RAI/Clase16/Python/Interface/Post.py
from Texto import Texto
from Imagen import Imagen
from Video import Video

class Post(Texto, Imagen, Video):

    def __init__(self, id:int, url:str, mensaje:str) -> None:
        self.id = id
        self.url = url
        self.mensaje = mensaje

    def ver_texto(self):
        print("> texto: " + self.mensaje)

    def ver_imagen(self):
        print("> Imagen: " + self.url)

    def ver_video(self):
        print("> Video: " + self.url)